py/vtdb/grpc_vtgate_client.py

```
@@ -0,0 +1,536 @@
+            tablet_type=topodata_pb2.TabletType.Value(tablet_type.upper()),
+            keyspace=keyspace_name,
+        )
+        _add_caller_id(request, effective_caller_id)
+        request.keyspace_ids.extend(keyspace_ids)
+        _convert_bind_vars(bind_variables, request.query.bind_variables)
+        it = self.stub.StreamExecuteKeyspaceIds(request, self.timeout)
+
+      elif keyranges is not None:
+        request = vtgate_pb2.StreamExecuteKeyRangesRequest(
+            query=query_pb2.BoundQuery(sql=sql),
+            tablet_type=topodata_pb2.TabletType.Value(tablet_type.upper()),
+            keyspace=keyspace_name,
+        )
+        _add_caller_id(request, effective_caller_id)
+        _add_key_ranges(request, keyranges)
+        _convert_bind_vars(bind_variables, request.query.bind_variables)
+        it = self.stub.StreamExecuteKeyRanges(request, self.timeout)
+
+      else:
+        request = vtgate_pb2.StreamExecuteRequest(
+            query=query_pb2.BoundQuery(sql=sql),
+            tablet_type=topodata_pb2.TabletType.Value(tablet_type.upper()),
+        )
+        _add_caller_id(request, effective_caller_id)
+        _convert_bind_vars(bind_variables, request.query.bind_variables)
+        it = self.stub.StreamExecute(request, self.timeout)
+
+      first_response = it.next()
+    except (face.AbortionError, vtgate_utils.VitessError) as e:
+      self.logger_object.log_private_data(bind_variables)
+      raise _convert_exception(
+          e, sql, keyspace_ids, keyranges,
+          keyspace=keyspace_name, tablet_type=tablet_type)
+
+    fields = []
+    conversions = []
+    for field in first_response.result.fields:
+      fields.append((field.name, field.type))
+      conversions.append(field_types_proto3.conversions.get(field.type))
+
+    def row_generator():
+      try:
+        for response in it:
+          for row in response.result.rows:
+            yield tuple(_make_row(row, conversions))
+      except Exception:
```

Any way we can pare this down to not catch all exceptions?